lib/fake_async.dart

```diff
@@ -12,284 +12,235 @@
-  ///
-  /// If [duration] is negative, the returned future completes with an
-  /// [ArgumentError].
-  ///
-  /// If a previous call to [elapse] has not yet completed, throws a
-  /// [StateError].
-  ///
-  /// Any Timers created within the zone used by [run] which are to expire
-  /// at or before the new time after [duration] has elapsed are run.
-  /// The microtask queue is processed surrounding each timer.  When a timer is
-  /// run, the [clock] will have been advanced by the timer's specified
-  /// duration.  Calls to [elapseBlocking] from within these timers and
-  /// microtasks which cause the [clock] to elapse more than the specified
-  /// [duration], can cause more timers to expire and thus be called.
-  ///
-  /// Once all expired timers are processed, the [clock] is advanced (if
-  /// necessary) to the time this method was called + [duration].
-  void elapse(Duration duration);
-
-  /// Simulates the synchronous passage of time, resulting from blocking or
-  /// expensive calls.
-  ///
-  /// Neither timers nor microtasks are run during this call.  Upon return, the
-  /// [clock] will have been advanced by [duration].
-  ///
-  /// If [duration] is negative, throws an [ArgumentError].
-  void elapseBlocking(Duration duration);
-
-  /// Runs [callback] in a [Zone] with fake timer and microtask scheduling.
-  ///
-  /// Uses
-  /// [ZoneSpecification.createTimer], [ZoneSpecification.createPeriodicTimer],
-  /// and [ZoneSpecification.scheduleMicrotask] to store callbacks for later
-  /// execution within the zone via calls to [elapse].
-  ///
-  /// Calls [callback] with `this` as argument and returns the result returned
-  /// by [callback].
-  dynamic run(callback(FakeAsync self));
-
-  /// Runs all remaining microtasks, including those scheduled as a result of
-  /// running them, until there are no more microtasks scheduled.
+/// The synchronous passage of time (as from blocking or expensive calls) can
+/// also be simulated using [elapseBlocking].
+class FakeAsync {
+  /// The amount of fake time that's elapsed since this [FakeAsync] was
+  /// created.
+  Duration get elapsed => _elapsed;
+  var _elapsed = Duration.ZERO;
+
+  /// The fake time at which the current call to [elapse] will finish running.
   ///
-  /// Does not run timers.
-  void flushMicrotasks();
+  /// This is `null` if there's no current call to [elapse].
+  Duration _elapsingTo;
 
-  /// Runs all timers until no timers remain (subject to [flushPeriodicTimers]
-  /// option), including those scheduled as a result of running them.
-  ///
-  /// [timeout] lets you set the maximum amount of time the flushing will take.
-  /// Throws a [StateError] if the [timeout] is exceeded. The default timeout
-  /// is 1 hour. [timeout] is relative to the elapsed time.
-  void flushTimers(
-      {Duration timeout: const Duration(hours: 1),
-      bool flushPeriodicTimers: true});
+  /// The queue of microtasks that are scheduled to run when fake time
```

Comment:
[nit] Avoid repeating type information in the doc comment. "The tasks that are scheduled to run when fake time progresses."

Reply:
Done.